doc/index.rst

```diff
@@ -0,0 +1,600 @@
+
+   In Python 3.12, the performance of runtime-checkable protocols was
+   improved, and ``typing_extensions`` backports these performance
+   improvements.
+
+Functions
+~~~~~~~~~
+
+.. function:: assert_never(arg)
+
+   See :py:func:`typing.assert_never`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.1.0
+
+.. function:: assert_type(val, typ)
+
+   See :py:func:`typing.assert_type`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.2.0
+
+.. function:: clear_overloads()
+
+   See :py:func:`typing.clear_overloads`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.2.0
+
+.. function:: get_args(tp)
+
+   See :py:func:`typing.get_args`. In ``typing`` since 3.8.
+
+   This function was changed in 3.9 and 3.10 to deal with :data:`Annotated`
+   and :class:`ParamSpec` correctly; ``typing_extensions`` backports these
+   fixes.
+
+.. function:: get_origin(tp)
+
+   See :py:func:`typing.get_origin`. In ``typing`` since 3.8.
+
+   This function was changed in 3.9 and 3.10 to deal with :data:`Annotated`
+   and :class:`ParamSpec` correctly; ``typing_extensions`` backports these
+   fixes.
+
+.. function:: get_original_bases(cls)
+
+   See :py:func:`types.get_original_bases`. Added to the standard library
+   in Python 3.12.
+
+   This function should always produce correct results when called on classes
+   constructed using features from ``typing_extensions``. However, it may
+   produce incorrect results when called on some :py:class:`NamedTuple` or
+   :py:class:`TypedDict` classes on Python <=3.11.
+
+   .. versionadded:: 4.6.0
+
+.. function:: get_overloads(func)
+
+   See :py:func:`typing.get_overloads`. In ``typing`` since 3.11.
+
+   Before Python 3.11, this works only with overloads created through
+   :func:`overload`, not with :py:func:`typing.overload`.
+
+   .. versionadded:: 4.2.0
+
+.. function:: get_type_hints(obj, globalns=None, localns=None, include_extras=False)
+
+   See :py:func:`typing.get_type_hints`.
+
+   In Python 3.11, this function was changed to support the new
+   :py:data:`typing.Required`. ``typing_extensions`` backports these
+   fixes.
+
+   .. versionchanged:: 4.1.0
+
+      Interaction with :data:`Required` and :data:`NotRequired`.
+
+.. function:: is_typeddict(tp)
+
+   See :py:func:`typing.is_typeddict`. In ``typing`` since 3.10.
+
+   On versions where :class:`TypedDict` is not the same as
+   :py:class:`typing.TypedDict`, this function recognizes
+   ``TypedDict`` classes created through either mechanism.
+
+   .. versionadded:: 4.1.0
+
+.. function:: reveal_type(obj)
+
+   See :py:func:`typing.reveal_type`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.1.0
+
+Other
+~~~~~
+
+.. class:: Text
+
+   See :py:class:`typing.Text`. In ``typing`` since 3.5.2.
+
+.. data:: TYPE_CHECKING
+
+   See :py:data:`typing.TYPE_CHECKING`. In ``typing`` since 3.5.2.
+
+Limitations
+-----------
+
+There are a few types whose interface was modified between different
```

Comment:
Or, if we keep "whose", then "interface" needs to be plural ("interfaces were" rather than "interface was")

```suggestion
There are a few types where the interface was modified between different
```